presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectStreamReader.java

```diff
@@ -121,12 +123,15 @@ public Block readBlock(Type type)
             }
         }
 
-        if (isNullVector.length < nextBatchSize) {
-            isNullVector = new boolean[nextBatchSize];
-        }
+        // create new isNullVector and offsetVector for VariableWidthBlock
+        boolean[] isNullVector = new boolean[nextBatchSize];
+        int[] offsetVector = new int[nextBatchSize + 1];
```

Comment:
You could re-use `offsetVector` as `lengthVector`. This would make the code a little bit hard to read. I guess it's better the way you wrote it.